.github/workflows/android.yml

```diff
@@ -1,118 +1,139 @@
 # This is a basic workflow to help you get started with Actions
 
-name: Android CI/CD v1
+name: Android release
 
 # Controls when the action will run. Triggers the workflow on push or pull request
 # events but only for the master branch
 on:
   push:
+    branches: [master, staging, next, 3493-code-push]
+  pull_request:
     branches: [master, staging, next]
+    types:
+      - closed
 
   workflow_dispatch:
+    inputs:
+      release:
+        description: 'release type'
+        required: true
+        default: ''
+        type: choice
+        options:
+          - dev: 'master'
+          - qa: 'staging'
+          - prod: 'next'
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  test:
-    name: Detox Tests
+  codepush:
+    name: Hot Code Push
     runs-on: ubuntu-latest
-    #disabled until we get it working again
-    if: ${{ false && !contains(github.event.head_commit.message, 'skip ci') }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
-        with:
-          node-version: 14
-      - name: Use specific Java version for sdkmanager to work
-        uses: joschi/setup-jdk@v1
-        with:
-          java-version: 'openjdk11'
-          architecture: 'x64'
-      - name: Install npm dependencies
+      - name: Pre-checks - Env is Dev
```

Comment:
those checks are the same for build & code push
we could move them to the reusable workflow: 
https://docs.github.com/en/actions/using-workflows/reusing-workflows

Reply:
also 'Cache & install npm dependencies' and 'add .env secrets'